src/data_structures/annot_map.rs

```diff
@@ -1,5 +1,25 @@
 //! Efficient container for locations annotated across a set of named
 //! reference sequences.
+//!
+//! # Example
+//!
+//! ```
+//! extern crate bio_types;
+//! use bio_types::strand::ReqStrand;
+//! use bio_types::annot::contig::Contig;
+//! use bio::data_structures::annot_map::AnnotMap;
+//!
+//! let mut genes: AnnotMap<String,String> = AnnotMap::new();
+//! let tma19 = Contig::new("chrXI".to_owned(), 334412, (334916 - 334412), ReqStrand::Reverse);
+//! let tma22 = Contig::new("chrX".to_owned(), 461829, 462426 - 461829, ReqStrand::Forward);
+//! genes.insert_at("TMA19".to_owned(), &tma19);
+//! genes.insert_at("TMA22".to_owned(), &tma22);
+//!
```

Comment:
Maybe also add an example of `insert_loc()`. This was previously missing.

Reply:
Sounds good, I will work on adding this

Reply:
Example added! Let me know what you think.

Reply:
Thanks for the good example!

Looking at it, I've finally developed a better understanding of this data structure and its methods, and might understand why there was no example for `insert_loc()`, so far. I tried simply adding `assert_eq!()` that makes sense and realised, that `insert_loc()` without an annotation data type that implements `Loc` but holds further information, doesn't really make much sense. But I kept the doctest for `insert_loc()`, just moving it to the function implementation itself.  As the main example, I only leave the `insert_at()`, which actually uses the location only for, well, the location to insert at and then actually inserts some useful extra information (the gene name `String`, here).

Also, I understand `insert_at()` a bit better, now, and added some description of its arguments that's hopefully more useful than what it said before? Please have a look if that matches your understanding, as you've spent some more time on this!

Reply:
Thanks, I think the descriptions are much clearer! Last thing I did was replace the `AnnotMap::new` doc comment example with a reference to the `insert_` functions given the types can vary.